python-checks/src/test/resources/checks/torchModuleShouldCallInit.py

```diff
@@ -0,0 +1,48 @@
+import torch.nn as nn
+
+class NonCompliantModule(nn.Module):
+     #^^^^^^^^^^^^^^^^^^> {{Inheritance happens here}}
+    def __init__(self): #Noncompliant {{Add a call to super().__init__()}}
+       #^^^^^^^^
+        ...
+
+class NonCompliantModule(nn.Module):
+    def __init__(self): #Noncompliant
```

Comment:
Are this and the next example FPs? Are we sure we want to raise in such a case?

Reply:
Hmm, in a way yes. Can we easily check if some where in the call graph super().__init__() is called?

Reply:
I think to minimize the FPs we could check if a call is made to `super().__init__()` anywhere in the class body. This would result in FNs that I think are acceptable.
```
class NonCompliantModule(OtherModule, nn.Module):
    def __init__(self): #FN
        not_call_super()

    def call_super(self):
        super().__init__()

class NonCompliantModule2(OtherModule, nn.Module):
    def __init__(self): #FN
        do_something()

    class Nested(Other):
        def __init__(self):
            super().__init__()
```